docs/source/tutorials/datasets.rst

```diff
@@ -108,33 +153,60 @@ Here is an example of sample that is formatted with :class:`~torchtune.data.Alpa
 
 We provide `other instruct templates <https://github.com/pytorch/torchtune/blob/main/torchtune/data/_instruct_templates.py>`_
 for common tasks such summarization and grammar correction. If you need to create your own
-instruct template for a custom task, you can create your own :class:`~torchtune.data.InstructTemplate`
-class and point to it in the config.
+instruct template for a custom task, you can inherit from :class:`~torchtune.data.InstructTemplate`
+and create your own class.
+
+.. code-block:: python
+
+    from torchtune.datasets import instruct_dataset
+    from torchtune.data import InstructTemplate
+
+    class CustomTemplate(InstructTemplate):
+        # Define the template as string with {} as placeholders for data columns
+        template = ...
+
+        # Implement this method
+        @classmethod
+        def format(
+            cls, sample: Mapping[str, Any], column_map: Optional[Dict[str, str]] = None
+        ) -> str:
+            ...
+
+    # Load in tokenizer
+    tokenizer = ...
+    dataset = instruct_dataset(
```

Comment:
Aside: we have both instruct/chat datasets and templates. Can they only go together? Do we mention this or enforce this?

Reply:
good point. we only enforce it through typing. but it might be good to do an explicit check.